sdk/storage/azure-storage-blob-changefeed/src/test/java/com/azure/storage/blob/changefeed/ChunkTest.groovy

```diff
@@ -0,0 +1,209 @@
+
+        mockContainer = mock(BlobContainerAsyncClient.class)
+        mockBlob = mock(BlobAsyncClient.class)
+        mockAvroReaderFactory = mock(AvroReaderFactory.class)
+        mockAvroReader = mock(AvroReader.class)
+        mockBlobLazyDownloaderFactory = mock(BlobLazyDownloaderFactory.class)
+        mockBlobLazyDownloader = mock(BlobLazyDownloader.class)
+
+        when(mockContainer.getBlobAsyncClient(anyString()))
+            .thenReturn(mockBlob)
+        when(mockBlobLazyDownloaderFactory.getBlobLazyDownloader(any(BlobAsyncClient.class), anyLong(), anyLong()))
+            .thenReturn(mockBlobLazyDownloader)
+        /* The data returned by the lazy downloader does not matter since we're mocking avroObjects.  */
+        when(mockBlobLazyDownloader.download())
+            .thenReturn(Flux.empty())
+        when(mockAvroReader.readAvroObjects())
+            .thenReturn(Flux.fromIterable(mockAvroObjects))
+    }
+
+    /* Tests no user cursor. */
+    def "getEvents min"() {
+        setup:
+        when(mockAvroReaderFactory.getAvroReader(any(Flux.class)))
+            .thenReturn(mockAvroReader)
+
+        when:
+        ChunkFactory factory = new ChunkFactory(mockAvroReaderFactory, mockBlobLazyDownloaderFactory, mockContainer)
+        Chunk chunk = factory.getChunk(chunkPath, chunkCursor, 0, 0)
+        def sv = StepVerifier.create(chunk.getEvents().index())
+
+        then:
+        sv.expectNextMatches({ tuple2 -> this.&verifyWrapper(tuple2.getT2(), tuple2.getT1(), 1234, 0) })
+            .expectNextMatches({ tuple2 -> this.&verifyWrapper(tuple2.getT2(), tuple2.getT1(), 1234, 1) })
+            .expectNextMatches({ tuple2 -> this.&verifyWrapper(tuple2.getT2(), tuple2.getT1(), 1234, 2) })
+            .expectNextMatches({ tuple2 -> this.&verifyWrapper(tuple2.getT2(), tuple2.getT1(), 1234, 3) })
+            .expectNextMatches({ tuple2 -> this.&verifyWrapper(tuple2.getT2(), tuple2.getT1(), 5678, 0) })
+            .expectNextMatches({ tuple2 -> this.&verifyWrapper(tuple2.getT2(), tuple2.getT1(), 5678, 1) })
+            .expectNextMatches({ tuple2 -> this.&verifyWrapper(tuple2.getT2(), tuple2.getT1(), 5678, 2) })
+            .expectNextMatches({ tuple2 -> this.&verifyWrapper(tuple2.getT2(), tuple2.getT1(), 5678, 3) })
+            .expectNextMatches({ tuple2 -> this.&verifyWrapper(tuple2.getT2(), tuple2.getT1(), 9101, 0) })
+            .expectNextMatches({ tuple2 -> this.&verifyWrapper(tuple2.getT2(), tuple2.getT1(), 9101, 1) })
+            .verifyComplete()
+
+        verify(mockContainer).getBlobAsyncClient(chunkPath) || true
+        verify(mockBlobLazyDownloaderFactory).getBlobLazyDownloader(mockBlob, ChunkFactory.DEFAULT_BODY_SIZE, 0) || true
+        verify(mockBlobLazyDownloader).download() || true
+        verify(mockAvroReaderFactory).getAvroReader(Flux.empty()) || true
+        verify(mockAvroReader).readAvroObjects() || true
```

Comment:
Why are you or'ing with true?

Reply:
Groovy doesnt work so well with Mockito https://stackoverflow.com/questions/16678126/simple-mockito-verify-works-in-junit-but-not-spock

Reply:
Shouldn't you `and` with true, though? It seems like these statements will always trivially evaluate to true

Reply:
it is a little weird. but since the || true comes after, if the verify step errors out indicating failure, it will still error out and not just "work".